### PR TITLE
executor: support plan replayer multi sqls

### LINKED ISSUE #37798: Support Plan Replayer for multi sqls
## Enhancement

Currently Plan Replayer only support dumps one sql instead of multi sql in a statement. We need to support Plan Replayer for multi sqls like following:

```sql
plan replayer dump explain <analyze> 'sqls.txt'
```

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close #37798 

Problem Summary:

### What is changed and how it works?

support plan replayer multi sqls

```sql
mysql> plan replayer dump explain 'sqls.txt';
Query OK, 0 rows affected (0.03 sec)

mysql> select @@tidb_last_plan_replayer_token;
+-----------------------------------------------------------+
| @@tidb_last_plan_replayer_token                           |
+-----------------------------------------------------------+
| replayer_LEDKg8sb-K0u24QesiH8ig==_1663226556509182000.zip |
+-----------------------------------------------------------+
1 row in set (0.00 sec)

```

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/plan_replayer.go [hunk 1]
```diff
@@ -61,101 +62,75 @@ const (
...
-	} else if s, ok := in.(*ast.SelectStmt); ok {
-		if s.With != nil && len(s.With.CTEs) > 0 {
-			for _, cte := range s.With.CTEs {
-				tne.cteNames[cte.Name.L] = struct{}{}
-			}
+			return err
 		}
+		e.endFlag = true
+		e.ctx.GetSessionVars().LastPlanReplayerToken = e.DumpInfo.FileName
+		return nil
```
Comment (Yisaer) [inline executor/plan_replayer.go:95]: As we can only read file from `handleSpecialQuery`, thus we store the file token in the seesion var return return nil here.

File: executor/plan_replayer.go [hunk 2]
```diff
@@ -146,16 +155,56 @@ func (e *PlanReplayerExec) Next(ctx context.Context, req *chunk.Chunk) error {
...
+	}
+	key := base64.URLEncoding.EncodeToString(b)
+	fileName := fmt.Sprintf("replayer_%v_%v.zip", key, time)
+	zf, err := os.Create(filepath.Join(path, fileName))
+	if err != nil {
+		return errors.AddStack(err)
+	}
+	e.DumpInfo.File = zf
+	e.DumpInfo.FileName = fileName
+	e.ctx.GetSessionVars().LastPlanReplayerToken = e.DumpInfo.FileName
```
Comment (xuyifangreeneyes) [inline executor/plan_replayer.go:207]: If some error happens when dumping info, do we need to clear `LastPlanReplayerToken`?

File: executor/plan_replayer.go [hunk 3]
```diff
@@ -644,14 +706,17 @@ type PlanReplayerLoadInfo struct {
 	Ctx  sessionctx.Context
 }
 
-type planReplayerLoadKeyType int
+type planReplayerKeyType int
 
-func (k planReplayerLoadKeyType) String() string {
+func (k planReplayerKeyType) String() string {
```
Comment (xuyifangreeneyes) [inline executor/plan_replayer.go:711]: It seems that we also need to change the function implementation.

File: executor/plan_replayer.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply executor/plan_replayer.go:169]: Maybe we can add the comment in code.

File: executor/plan_replayer.go (hunk 2 above)
Comment (Yisaer) [reply executor/plan_replayer.go]: updated.

**Review by hawkingrei — APPROVED** [review]

File: executor/plan_replayer.go [hunk 4]
```diff
@@ -644,6 +699,12 @@ type PlanReplayerLoadInfo struct {
 	Ctx  sessionctx.Context
 }
 
+type planReplayerDumpVarKey int
```
Comment (xuyifangreeneyes) [inline executor/plan_replayer.go:702]:
```suggestion
type planReplayerDumpKeyType int
```

Comment (xuyifangreeneyes) [reply executor/plan_replayer.go:702]: Maybe it is more consistent.

**Review by xuyifangreeneyes — APPROVED** [review]

**Pull request merged**